lib/housing-available/housing-available.controllers.js

```diff
@@ -1 +1,209 @@
-module.exports = {};
\ No newline at end of file
+module.exports = (models) => {
+  const Boom = require('boom');
+  const Sequelize = require('sequelize');
+  const Op = Sequelize.Op;
+  // const housingAvailableModels = require('./housing-available.models');
+  // const housingAvailableRoutes = require('./housing-available.routes');
+  const Log = require('../log/log.controllers')(models).lib;
+  const accountLib = require('../account/account.controllers')(models).lib;
+  const propertyLib = require('../property/property.controllers')(models).lib;
+  const locationLib = require('../location/location.controllers')(models).lib;
+  const postalAddressLib = require("../postal-address/postal-address.controllers")(models).lib;
+
+  return {
+
+    // Create new housingAvailable rental listing;
+    postHousingAvailable: async function (request, h) {
+      /*
+        # Require Account; user must be a logged-in user (i.e. have an Account) to post a rental;
+      */
+      let account;
+      let payload = request.payload;
+      try {
+        account = await accountLib.getAccount(request.auth.credentials);
+      } catch (e) {
+        throw Boom.badImplementation('Error during getAccount(request.auth.credentials). ' + e);
+      }
+
+      if (account === null) {
+        throw Boom.badRequest('Must have an Account to create a HousingAvailable');
+      }
+
+      /* Restrict AuthorId to same user, or SuperAdmin */
+      if (payload.AuthorId) {
+        // If AuthorId is set then we have already passed auth and account lookup
+        // This is because if AuthorId is set then not only quickInfo is set
+        if (
+          payload.AuthorId !== account.id &&
+          request.auth.credentials.subjectId !== process.env.SUPER_ADMIN
+        ) {
+          throw Boom.badRequest('HousingAvailable.AuthorId must match your Account ID or you must be a Super Admin');
+        }
+      }
+
+      // bedrooms validation:
+      let hasBedrooms = false;
+      let bedrooms = Number(request.payload.bedrooms);
+      if (request.payload.hasOwnProperty("bedrooms")) {
+        hasBedrooms = true;
+      }
+
+      // bathrooms validation:
+      let hasBathrooms = false;
+      let bathrooms = Number(request.payload.bathrooms);
+      if (request.payload.hasOwnProperty("bathrooms")) {
+        hasBathrooms = true;
+      }
+
+      // bathrooms can be in .25 increments
+      if (hasBathrooms && !bathroomsValidIncrement(bathrooms)) {
+        throw Boom.badData('bathrooms must be in .25 increments');
+      }
+
+      // start building HousingAvailable object
+      let housingAvailableObject = {};
+      if (request.payload.title && request.payload.title !== request.payload.address) {
+        housingAvailableObject.title = request.payload.title;
+      }
+
+      if (request.payload.AuthorId) {
+        housingAvailableObject.AuthorId = request.payload.AuthorId;
+      } else {
+        housingAvailableObject.AuthorId = request.payload.account.id;
```

Comment:
There's no account object on request.payload. The account I was referring to was the one you initialized on line 20 and then filled in the few lines after that, and then ensured it had a result after that.

so the right side of that expression just needs to be account.id;